hadoop-ozone/dist/src/main/compose/ozones3-haproxy/docker-compose.yaml

```diff
@@ -0,0 +1,83 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: "3"
+services:
+   s3g:
+      image: haproxy:latest
+      volumes:
+         - ../..:/opt/hadoop
+         - ./haproxy-conf/haproxy.cfg:/usr/local/etc/haproxy/haproxy.cfg
+      ports:
+         - 9878:9878
+   datanode:
+      image: apache/ozone-runner:${HADOOP_RUNNER_VERSION}
+      volumes:
+        - ../..:/opt/hadoop
+      ports:
+        - 9864
+      command: ["ozone","datanode"]
+      env_file:
+        - ./docker-config
+   om:
+      image: apache/ozone-runner:${HADOOP_RUNNER_VERSION}
+      volumes:
+         - ../..:/opt/hadoop
+      ports:
+         - 9874:9874
+      environment:
+         ENSURE_OM_INITIALIZED: /data/metadata/om/current/VERSION
+      env_file:
+          - ./docker-config
+      command: ["ozone","om"]
+   scm:
+      image: apache/ozone-runner:${HADOOP_RUNNER_VERSION}
+      volumes:
+         - ../..:/opt/hadoop
+      ports:
+         - 9876:9876
+      env_file:
+          - ./docker-config
+      environment:
+          ENSURE_SCM_INITIALIZED: /data/metadata/scm/current/VERSION
+      command: ["ozone","scm"]
+   s3g1:
```

Comment:
These can be collapsed into one entry, then we can use scale option. Not a blocker to commit.

Reply:
Yes, we can do that, but we need to have ha-proxy image which can have some configuration, where we can dynamically configure s3proxy and start/stop.